pkg/scheduler/context.go

```diff
@@ -640,10 +661,21 @@ func (cc *ClusterContext) updateNode(nodeInfo *si.NodeInfo) {
 	case si.NodeInfo_DRAIN_NODE:
 		// set the state to not schedulable
 		node.SetSchedulable(false)
+		metrics.GetSchedulerMetrics().IncDrainingNodes()
```

Comment:
oops, I overlooked an error here I think: if the node is already draining we should not increment again and we do not need to change state:
```
if node.IsSchedulable() {
	node.SetSchedulable(false)
	metrics.GetSchedulerMetrics().IncDrainingNodes()
}
```
this is handled the other way around

Reply:
Fixed